src/core/ChannelHandler.js

```diff
@@ -0,0 +1,103 @@
+import Observable from 'zen-observable';
+
+const events = [
+  'phx_reply',
+  'phx_error',
+  'status_changed',
+  'stopped',
+  'error',
+  'failed',
+  'starting',
+  'started',
+  'deploying',
+  'deployed',
+  'deployment_failed',
+  'ready',
+  'terminated',
+  'snapshot_taken',
+  'snapshot_reverted'
+];
+
+export default class ChannelHandler {
+  constructor(name, socket, globalLogger) {
+    this._name = name;
+    this._channel = socket.channel(this._name);
+    this._stream = this._buildChannelStream(events);
+    this._log = globalLogger.extend(`${this._name}`);
+    this._logger = this._stream.subscribe(value =>
+      this._log(JSON.stringify(value, null, 4))
+    );
+    this.init();
+  }
+
+  _buildChannelStream(eventsList) {
+    return new Observable(stream => {
+      const setChannelEvent = event => {
+        this._channel.on(event, payload => {
+          switch (event) {
+            case 'deployment_failed':
+              stream.error({ event, payload });
```

Comment:
when you call `stream.error` does it keep the subscription open for future events?
Also, not sure if `deployment_failed` should be hardcoded as a stream error, it's a status update about a failed testchain deployment?

Reply:
The easiest way to explain is via [this example](https://github.com/zenparsing/zen-observable#observablesubscribenextcallback-errorcallback-completecallback).
It doesn't necessarily throw an error but leaves that for the user to handle at the error callback. It does keep that subscription open.

Will change `deployment_failed`